opentracing-spring-cloud/src/main/java/io/opentracing/contrib/spring/cloud/async/CustomAsyncConfigurerAutoConfiguration.java

```diff
@@ -0,0 +1,45 @@
+package io.opentracing.contrib.spring.cloud.async;
+
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.BeanPostProcessor;
+import org.springframework.boot.autoconfigure.AutoConfigureBefore;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnBean;
+import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.scheduling.annotation.AsyncConfigurer;
+
+import io.opentracing.Tracer;
+import io.opentracing.contrib.spring.cloud.async.instrument.TracedAsyncConfigurer;
+
+/**
+ * {@link org.springframework.boot.autoconfigure.EnableAutoConfiguration Auto-configuration}
+ * that wraps an existing custom {@link AsyncConfigurer} in a {@link TracedAsyncConfigurer}
+ *
+ * @author Dave Syer
+ * @author Pavol Loffay
+ */
+@Configuration
+@ConditionalOnBean(AsyncConfigurer.class)
+//TODO when Scheduling is added we need to do @AutoConfigurationAfter on it
+@AutoConfigureBefore(DefaultAsyncAutoConfiguration.class)
+@ConditionalOnProperty(name = "opentracing.spring.cloud.async.enabled", havingValue = "true", matchIfMissing = true)
```

Comment:
Is this property name too generic - will all async related auto configs be enabled/disabled based on this single property, or should it be made more specific to the scheduling?

Reply:
Package name implies this is the auto-config for 'async' functionality in spring-cloud so possibly ok - but if decide this is more specific to 'scheduling async' then may also need to make the package and possibly artifact id more specific to scheduling.

Reply:
this PR does not deal with scheduling. Scheduling will be added in a separate PR with its own auto-config and conditional property.